presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationMetadata.java

```diff
@@ -116,12 +117,14 @@ public MethodHandle getOutputFunction()
     private static void verifyInputFunctionSignature(MethodHandle method, List<ParameterMetadata> parameterMetadatas, Class<?> stateInterface)
     {
         Class<?>[] parameters = method.type().parameterArray();
-        checkArgument(stateInterface == parameters[0], "First argument of aggregation input function must be %s", stateInterface.getSimpleName());
         checkArgument(parameters.length > 0, "Aggregation input function must have at least one parameter");
-        checkArgument(parameterMetadatas.get(0).getParameterType() == STATE, "First parameter must be state");
-        for (int i = 1; i < parameters.length; i++) {
+        checkArgument(parameterMetadatas.stream().filter(m -> m.getParameterType() == STATE).count() == 1, "There must be exactly one state parameter in input function");
+        for (int i = 0; i < parameters.length; i++) {
             ParameterMetadata metadata = parameterMetadatas.get(i);
             switch (metadata.getParameterType()) {
+                case STATE:
+                    checkArgument(stateInterface == parameters[i], String.format("State argument must be of type %s", stateInterface));
+                    break;
```

Comment:
What is the incentive behind relaxing the requirement that `state` must be the first argument?